### PR TITLE
#184651343 implement functionality for viewing specific items as a buyer or seller

### PR DESCRIPTION
**Pull request by bahati10**
### **What does this PR do?**
A User should be able to view a specific item
##### **Description of Task to be completed?**
As a User
I should be able to retrieve a given item.
  **As a Seller** I should be able to see an item if it is from my 
   collection

  **As a Buyer**, I should be able to see any item from the available items.

#### How should this be manually tested?
- First Clone the repository  
- Navigate to the branch project 
- use **`npm install`** to install the required dependencies.
- to test use **`npm run test`** and see the coverage.
- use **`npm start`** to start the server
- You can log in as a seller to list all items in your collection
- Or you can log in as a buyer to see all available products

#### What are the relevant pivotal tracker/Trello stories?
[#184651324](https://www.pivotaltracker.com/n/projects/2633635)

### CODE REVIEW

File: src/controllers/findProduct.controller.js
Comment (nshutip) [inline src/controllers/findProduct.controller.js:1]: - Why do have have a try/catch block in your controller, while you imported asyncWrapper to handle it?
- All validation should be handled by a middleware.

**Review by nshutip — DISMISSED** [review]
Functionality well implemented, but you need to make some changes as mentioned in the other comments.

**Review by ruberwa — DISMISSED** [review]

**Review by ruberwa — DISMISSED** [review]
Working good!

**Review by JulesNIYOMWUNGERI — DISMISSED** [review]
working well!!

**Review by nshutip — DISMISSED** [review]

**Review by ruberwa — DISMISSED** [review]

**Review by ruberwa — APPROVED** [review]

**Review by nshutip — APPROVED** [review]

**Review by JulesNIYOMWUNGERI — APPROVED** [review]

**Review by kagame8732 — APPROVED** [review]

**Pull request merged**